fearless/ApplicationLayer/Services/Balance/WalletBalanceSubscription/WalletBalanceSubscriptionAdapter.swift

```diff
@@ -312,34 +305,18 @@ extension WalletBalanceSubscriptionAdapter: EventVisitorProtocol {
         pricesProvider?.refresh()
     }
 
+    func processSelectedAccountChanged(event _: SelectedAccountChanged) {
+        fetchAllMetaAccounts()
+    }
+
     func processChainsUpdated(event: ChainsUpdatedEvent) {
         let updatedChainAssets = event.updatedChains.map(\.chainAssets).reduce([], +)
 
         updatedChainAssets.forEach { chainAsset in
             let key = chainAsset.chainAssetId
             self.chainAssets[key] = chainAsset
         }
-        buildBalance()
-    }
-
-    func processSelectedAccountChanged(event: SelectedAccountChanged) {
-        if let index = metaAccounts.firstIndex(where: { $0.metaId == event.account.metaId }) {
-            metaAccounts[index] = event.account
-        }
-
-        switch subscriptionType {
-        case .wallets:
-            reset()
-            fetchAllMetaAccounts()
-        case let .wallet(walletId):
-            reset()
-            fetchMetaAccount(by: walletId, chainAsset: nil)
-        case let .chainAsset(walletId, chainAsset):
-            reset()
-            fetchMetaAccount(by: walletId, chainAsset: chainAsset)
-        case .none:
-            break
-        }
+        notifyIfNeeded(with: metaAccounts, updatedChainAssets: updatedChainAssets)
```

Comment:
If we added new chains on remote json, this won't work. You add new chainAssets in local array, but not subscribe on the account infos and not recalculate expected account infos